vispy/scene/widgets/widget.py

```diff
@@ -115,6 +118,79 @@ def size(self, s):
         self.events.resize()
 
     @property
+    def width(self):
+        """The width of this widget"""
+        return self._size[0]
+
+    @width.setter
+    def width(self, width):
+        self.size = (width, self._size[1])
+
+    @property
+    def min_width(self):
+        return self._width_limits[0]
+
+    @min_width.setter
+    def min_width(self, min_width):
+        if min_width is None:
+            self._width_limits[0] = None
+            return
+
+        self._width_limits[0] = float(min_width)
+
+    @property
+    def max_width(self):
+        return self._width_limits[1]
+
+    @max_width.setter
+    def max_width(self, max_width):
+        if max_width is None:
+            self._width_limits[1] = None
+            return
+
+        self._width_limits[1] = float(max_width)
+
+    def fixed_width(self, fixed_width):
+        self.min_width = self.max_width = fixed_width
+
+    @property
+    def height(self):
+        """The height of this widget"""
+        return self._size[1]
+
+    @height.setter
+    def height(self, height):
+        self.size = (self._size[0], height)
```

Comment:
I think if someone sets a `height`, what it should really do is set a fixed size